lib/modules/dosomething/dosomething_reportback/includes/Reportback.php

```diff
@@ -162,8 +162,11 @@ private function build($data, $full = false) {
 
     if ($full) {
       $northstar_user = dosomething_northstar_get_northstar_user($data->uid);
-      $northstar_user = json_decode($northstar_user, true);
-      $northstar_user = (object) $northstar_user['data'][0];
+
+      if (!empty($northstar_user)) {
+        $northstar_user = json_decode($northstar_user, true);
+        $northstar_user = (object) $northstar_user['data'][0];
+      }
 
       $this->user = [
```

Comment:
Does this need some sort of logic too? So if `$northstar_user` ends up being `null`, how is the `id`, `first_name`, `last_name` etc being assigned that depends on getting an object from NS and not getting `null`?